include/zephyr/bluetooth/conn.h

```diff
@@ -413,6 +413,109 @@ struct bt_conn_le_cs_fae_table {
 	uint8_t *remote_fae_table;
 };
 
+/** Procedure done status */
+enum bt_conn_le_cs_procedure_done_status {
+	BT_CONN_LE_CS_PROCEDURE_COMPLETE = BT_HCI_LE_CS_PROCEDURE_DONE_STATUS_COMPLETE,
+	BT_CONN_LE_CS_PROCEDURE_INCOMPLETE = BT_HCI_LE_CS_PROCEDURE_DONE_STATUS_PARTIAL,
+	BT_CONN_LE_CS_PROCEDURE_ABORTED = BT_HCI_LE_CS_PROCEDURE_DONE_STATUS_ABORTED,
+};
+
+/** Subevent done status */
+enum bt_conn_le_cs_subevent_done_status {
+	BT_CONN_LE_CS_SUBEVENT_COMPLETE = BT_HCI_LE_CS_SUBEVENT_DONE_STATUS_COMPLETE,
+	BT_CONN_LE_CS_SUBEVENT_INCOMPLETE = BT_HCI_LE_CS_SUBEVENT_DONE_STATUS_PARTIAL,
+	BT_CONN_LE_CS_SUBEVENT_ABORTED = BT_HCI_LE_CS_SUBEVENT_DONE_STATUS_ABORTED,
+};
+
+/** Procedure abort reason */
+enum bt_conn_le_cs_procedure_abort_reason {
+	BT_CONN_LE_CS_PROCEDURE_NOT_ABORTED = BT_HCI_LE_CS_PROCEDURE_ABORT_REASON_NO_ABORT,
+	BT_CONN_LE_CS_PROCEDURE_ABORT_REQUESTED =
+		BT_HCI_LE_CS_PROCEDURE_ABORT_REASON_LOCAL_HOST_OR_REMOTE_REQUEST,
+	BT_CONN_LE_CS_PROCEDURE_ABORT_TOO_FEW_CHANNELS =
+		BT_HCI_LE_CS_PROCEDURE_ABORT_REASON_TOO_FEW_CHANNELS,
+	BT_CONN_LE_CS_PROCEDURE_ABORT_CHMAP_INSTANT_PASSED =
+		BT_HCI_LE_CS_PROCEDURE_ABORT_REASON_CHMAP_INSTANT_PASSED,
+	BT_CONN_LE_CS_PROCEDURE_ABORT_UNSPECIFIED = BT_HCI_LE_CS_PROCEDURE_ABORT_REASON_UNSPECIFIED,
+};
+
+/** Subevent abort reason */
+enum bt_conn_le_cs_subevent_abort_reason {
+	BT_CONN_LE_CS_SUBEVENT_NOT_ABORTED = BT_HCI_LE_CS_SUBEVENT_ABORT_REASON_NO_ABORT,
+	BT_CONN_LE_CS_SUBEVENT_ABORT_REQUESTED =
+		BT_HCI_LE_CS_SUBEVENT_ABORT_REASON_LOCAL_HOST_OR_REMOTE_REQUEST,
+	BT_CONN_LE_CS_SUBEVENT_ABORT_NO_CS_SYNC =
+		BT_HCI_LE_CS_SUBEVENT_ABORT_REASON_NO_CS_SYNC_RECEIVED,
+	BT_CONN_LE_CS_SUBEVENT_ABORT_SCHED_CONFLICT =
+		BT_HCI_LE_CS_SUBEVENT_ABORT_REASON_SCHED_CONFLICT,
+	BT_CONN_LE_CS_SUBEVENT_ABORT_UNSPECIFIED = BT_HCI_LE_CS_SUBEVENT_ABORT_REASON_UNSPECIFIED,
+};
+
+/** Subevent data for LE connections supporting CS */
+struct bt_conn_le_cs_subevent_result {
+	struct {
+		/** CS configuration identifier.
+		 *
+		 *  Range: 0 to 3
+		 *
+		 *  If these results were generated by a CS Test,
+		 *  this value will be set to 0 and has no meaning.
+		 */
+		uint8_t config_id;
+		/** Starting ACL connection event counter.
+		 *
+		 *  If these results were generated by a CS Test,
+		 *  this value will be set to 0 and has no meaning.
+		 */
+		uint16_t start_acl_conn_event;
+		/** CS procedure count associated with these results.
+		 *
+		 *  This is the CS procedure count since the completion of
+		 *  the Channel Sounding Security Start procedure.
+		 */
+		uint16_t procedure_counter;
+		/** Frequency compensation value in units of 0.01 ppm.
+		 *
+		 *  This is a 15-bit signed integer in the range [-100, 100] ppm.
+		 *
+		 *  A value of @ref BT_HCI_LE_CS_SUBEVENT_RESULT_FREQ_COMPENSATION_NOT_AVAILABLE
+		 *  indicates that the role is not the initiator, or that the
+		 *  frequency compensation value is unavailable.
+		 */
+		uint16_t frequency_compensation;
```

Comment:
Should we report this in a more user friendly format ie. int16_t? Or we want to keep it raw in case the user algorithm uses it in the raw format?

Reply:
Thought process: I saw the RAS spec wants it like this 
![format freq compensation](https://github.com/user-attachments/assets/3f50f727-c8f3-4e49-b5d8-e491648d6873)